### PR TITLE
Initial point

### PR DESCRIPTION
**Pull request by svkrdj**
Split Fuzzdb zaproxy/zaproxy [#5294](https://github.com/zaproxy/zaproxy/issues/5294)

### CODE REVIEW

**Comment by thc202** [comment]
The Gradle wrapper scripts (`gradlew` and `gradlew.bat`) need to be added as well (I forgot to include those). (Note that the wrapper JAR was not yet added.)

The `.gitignore` file should be updated to:
```
# Gradle
#-------
.gradle
/build
/buildSrc/build

# IDEA
# ----
.idea
.shelf
/*.iml
/*.ipr
/*.iws
/buildSrc/*.iml
/buildSrc/*.ipr
/buildSrc/*.iws
/buildSrc/out
/out

# Eclipse
# -------
*.classpath
*.project
*.settings
/bin
/buildSrc/bin

# NetBeans
# --------
.nb-gradle
.nb-gradle-properties
```
and the Eclipse specific files removed.

The backdoor files should be under `src/main/zapHomeFiles/fuzzers/fuzzdb/web-backdoors/` (like in the `zap-extensions` repo).

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+eclipse {
+    classpath {
+        // Prevent compilation of zapHomeFiles.
+        sourceSets = listOf()
+    }
+}
+
+version = "1"
+description = "FuzzDB web backdoors which can be used with the ZAP fuzzer"
+
```
Comment (thc202) [inline build.gradle.kts]: Lets insert here:
```kts
java {
    sourceCompatibility = JavaVersion.VERSION_1_8
    targetCompatibility = JavaVersion.VERSION_1_8
}
```
(The add-on doesn't have Java source, but better specify it anyway.)

**Comment by svkrdj** [comment]
How are we going to test it and use with zap?

**Comment by kingthorin** [comment]
Build it. Launch zap and load the freshly built addon? (File > Load Addon)

**Comment by svkrdj** [comment]
Build successfully and loaded in ZAP.

**Comment by kingthorin** [comment]
This looks good to me. Could you:

1. Normalize the line endings in all these files? (LF or `/n` [NOT CRLF]) As well as adding a `.gitattributes` file, ex: https://github.com/zaproxy/zap-hud/blob/develop/.gitattributes
2. Open a PR removing the files from the zap-extensions repo. (We can co-ordinate the release of the add-ons.)

**Comment by svkrdj** [comment]
@kingthorin  Sure, on my way. Do I have to normalize the file in directory `web-backdoors` also?

**Comment by kingthorin** [comment]
To answer your question (though it seems to have been removed/deleted), yes you can normalize the backdoor as well.

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+*.pl text
+*.cgi text
+*.cfm text
+*.gradle text
+*.gradle.kts text
+*.properties text
+*.md text
+
+*.bat text eol=crlf
+*.sh=text
```
Comment (kingthorin) [inline .gitattributes]: I believe this should be a space not an equals sign.

Comment (svkrdj) [reply .gitattributes]: Oh sorry, my mistake

**Comment by svkrdj** [comment]
I hope it's done. Let me know if any changes required. Meanwhile, I will open the PR in the zap-extensions repo.

**Comment by thc202** [comment]
The `README.md` (that was in `web-backdoors` dir) should be added as well.

There are some files that need to be normalised:
https://help.github.com/en/articles/dealing-with-line-endings#refreshing-a-repository-after-changing-line-endings

**Comment by thc202** [comment]
Sorry, I meant adding the `README.md` to its directory (`src/main/zapHomeFiles/fuzzers/fuzzdb/web-backdoors/`), we should add one to top level dir describing this project but can be done in a follow up PR.

**Comment by svkrdj** [comment]
I hope the line ending is properly configured let me I know if not (I haven't done this before so sorry for the silly mistakes :neutral_face:).

**Comment by thc202** [comment]
Looks great, thank you!

**Pull request merged**